storage.tf

```diff
@@ -17,11 +17,13 @@ locals {
           vol_name       = "${var.prefix}-${substr(var.subnets[subnet].id, -4, 4)}-${format("%03d", count + 1)}-${volume.name}"
           zone           = var.subnets[subnet].zone
           profile        = volume.profile
-          capacity       = volume.capacity
+          capacity       = (volume.snapshot_id == null) ? volume.capacity : null
           vsi_name       = "${var.subnets[subnet].name}-${count}"
-          iops           = volume.iops
-          encryption_key = var.kms_encryption_enabled ? var.boot_volume_encryption_key : volume.encryption_key
+          iops           = (volume.snapshot_id == null) ? volume.iops : null
+          encryption_key = (volume.snapshot_id == null) ? (var.kms_encryption_enabled ? var.boot_volume_encryption_key : volume.encryption_key) : null
```

Comment:
I guess this logic should be handled differently,
In a case where `kms_encryption_enabled` is true, at the same time the user wants to encrypt a particular volume with a different `encryption_key`.  Since `kms_encryption_enabled` is true the `encryption_key`  will have the value of `boot_volume_encryption_key` no matter the value user passes for `encryption_key`.

Reply:
Created a new issue to track this [#10196](https://github.ibm.com/GoldenEye/issues/issues/10196)

Reply:
We have created an issue for this and will handle in a different PR.